api/helpers.go

```diff
@@ -71,3 +73,28 @@ func NotFoundError(w http.ResponseWriter, message string) {
 func UnauthorizedError(w http.ResponseWriter, message string) {
 	sendJSON(w, 401, &HTTPError{Code: 401, Message: message})
 }
+
+func calculateTotalPages(perPage, total uint64) uint64 {
+	pages := total / perPage
+	if total%perPage > 0 {
+		return pages + 1
+	}
+	return pages
+}
+
+func addPaginationHeaders(w http.ResponseWriter, r *http.Request, page, perPage, total uint64) {
+	totalPages := calculateTotalPages(perPage, total)
+	url, _ := url.ParseRequestURI(r.URL.String())
+	query := url.Query()
+	header := ""
+	if totalPages > page {
+		query.Set("page", fmt.Sprintf("%v", page+1))
+		url.RawQuery = query.Encode()
+		header += "<" + url.String() + ">; rel=\"next\", "
+	}
+	query.Set("page", fmt.Sprintf("%v", totalPages))
```

Comment:
doesn't this need to be in a else block? It will override the query param 'page' right?